nnvm/python/nnvm/frontend/tensorflow.py

```diff
@@ -648,6 +648,26 @@ def _impl(inputs, attr, params):
             ignores=['Tpaddings'],)(new_inputs, attr)
     return _impl
 
+def _transpose():
+    def _impl(inputs, attr, params):
+        # If perm is not specified, axes is left empty,
+        # otherwise its value is get from params
+        param_name = inputs[1].list_output_names()[0]
+        axes = params.get(param_name, tvm.nd.array([])).asnumpy()
+        return _sym.transpose(inputs[0], axes=tuple(axes))
+    return _impl
+
+def _range():
+    def _impl(inputs, attr, params):
+        print("Range: It's a pass through, it is introduced by tf.transpose() but can be ignored")
+        return inputs[0]
+    return _impl
+
+def _rank():
```

Comment:
Output of Rank can be a param symbol. '_input_shapes' attribute will be help to find rank here.